### PR TITLE
migrate to viem

### PR DESCRIPTION
**Pull request by jhd2best**
test:unit
image

test:integration
image

### CODE REVIEW

File: packages/core-sdk/src/resources/collect.ts [hunk 1]
```diff
@@ -1,17 +1,17 @@
...
 
 import { CollectIPAssetRequest, CollectIPAssetResponse } from "../types/resources/collect";
-import { CollectModule } from "../abi/generated";
 import { handleError } from "../utils/errors";
 import { CollectReadOnlyClient } from "./collectReadOnly";
+import { collectModuleConfig } from "../abi/collectModule.abi";
 
 export class CollectClient extends CollectReadOnlyClient {
-  private readonly signer: Signer;
+  protected readonly wallet: WalletClient;
```
Comment (edisonz0718) [inline packages/core-sdk/src/resources/collect.ts:10]: Is the change from private to protected for testing?

File: packages/core-sdk/src/resources/relationship.ts [hunk 2]
```diff
@@ -35,19 +34,24 @@ export class RelationshipClient extends RelationshipReadOnlyClient {
    * @returns An object containing the source and destination IP registry addresses.
    */
   private async getRegistryAddresses(
-    sourceFranchiseId: string,
-    destFranchiseId: string,
+    sourceFranchiseId: bigint,
+    destFranchiseId: bigint,
```
Comment (DonFungible) [inline packages/core-sdk/src/resources/relationship.ts:38]: should these be string too? @edisonz0718

Comment (edisonz0718) [reply packages/core-sdk/src/resources/relationship.ts:38]: yeah all the ids should be string

File: packages/tsconfig/base.json [hunk 3]
```diff
@@ -15,7 +15,7 @@
     "noUnusedParameters": false,
     "skipLibCheck": true,
     "strict": true,
-    "lib": ["ESNext"]
+    "lib": ["ESNext", "dom"]
```
[the remark this answers is not in the document]
Comment (LuttyYang) [reply packages/tsconfig/base.json]: It was initially used by the script, but was later no longer needed. It was forgotten to be removed, and has now been removed.

File: packages/core-sdk/package.json [hunk 4]
```diff
@@ -26,15 +26,14 @@
     "lint:fix": "pnpm run lint --fix",
     "lint": "eslint ./src",
     "tsc": "tsc --noEmit",
-    "abigen": "typechain --target ethers-v5 './src/abi/**/*.json' --out-dir ./src/abi/generated"
+    "abigen": "ts-node-script scripts/abi-helper.ts"
```
[the remark this answers is not in the document]
Comment (LuttyYang) [reply packages/core-sdk/package.json]: It was initially used by the script, but was later no longer needed. It was forgotten to be removed, and has now been removed.

File: packages/core-sdk/src/types/resources/collect.ts [hunk 5]
```diff
@@ -15,8 +15,8 @@ export type Collection = {
  * @public
  */
 export type CollectIPAssetRequest = {
-  franchiseId: string;
-  ipAssetId: string;
+  franchiseId: bigint;
```
[the remark this answers is not in the document]
Comment (LuttyYang) [reply packages/core-sdk/src/types/resources/collect.ts]: The ID has now been changed to a string type.

File: packages/core-sdk/src/resources/relationship.ts (hunk 2 above)
Comment (LuttyYang) [reply packages/core-sdk/src/resources/relationship.ts]: The ID has now been changed to a string type.

File: packages/core-sdk/src/resources/collect.ts (hunk 1 above)
Comment (LuttyYang) [reply packages/core-sdk/src/resources/collect.ts]: typo, fixed

File: packages/core-sdk/src/abi/franchiseRegistry.abi.ts [hunk 6]
```diff
@@ -1,15 +1,12 @@
-/* Autogenerated file. Do not edit manually. */
```
[the remark this answers is not in the document]
Comment (LuttyYang) [reply packages/core-sdk/src/abi/franchiseRegistry.abi.ts:1]: This is not auto-generated, Viem does not need to generate code, ABI is the type

File: packages/core-sdk/src/client.ts [hunk 7]
```diff
@@ -54,9 +52,28 @@ export class StoryClient {
...
+
+      this.wallet = createWalletClient({
+        chain: (this.config as StoryConfig).chain || mainnet,
+        transport:
+            (this.config as StoryConfig).transport || http("https://cloudflare-eth.com"),
+        account: account,
+      });
+    } else {
+      this.rpcClient = createPublicClient(clientConfig);
+    }
```
[the remark this answers is not in the document]
Comment (LuttyYang) [reply packages/core-sdk/src/client.ts]: Writing it this way, the type of inputConfig cannot be inferred, leading to an error with account. Currently, the writing of this part has been optimized.

[the remark this answers is not in the document]
Comment (jhd2best) [reply packages/core-sdk/src/client.ts]: default chain and  rpc provider updated

**Pull request closed without merging**